### PR TITLE
Fix issue: showing error when deleting the board

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary
This fixes a bug where deleting a board showed the wrong screen. System admins got a "Join private board" pop-up, and regular users saw "Something went wrong", instead of being taken to the next available board or the template picker. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-66320

## Change Impact: 🟠 Medium

**Regression Risk:** Changes touch the board deletion and post-deletion navigation flows in two user-facing components (sidebarCategory and boardPage), adding logic to compute the next target board, clear persisted lastBoardId, and short-circuit re-verification for websocket-deleted boards. New tests cover many cases, but timing between websocket events, async API calls (octoClient), and navigation introduces moderate risk—edge cases (rapid deletions, category-scoped board selection, hidden/template boards) could still behave unexpectedly.

**QA Recommendation:** Perform manual QA focused on deletion flows across roles and board configurations: delete a board when there are adjacent boards, only earlier boards, boards in other categories, the last board in a team, and template boards. Verify admins do not see "Join private board" after deleting, regular users do not see "Something went wrong", and navigation goes to the expected board or template picker. Also test websocket-triggered deletions and rapid successive deletions to ensure no stale navigation or rejoin attempts occur.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: webapp/src/components/sidebar/sidebarCategory.tsx [hunk 1]
```diff
@@ -187,26 +189,60 @@ const SidebarCategory = (props: Props) => {
...
 
-                if (nextBoardId) {
-                // This delay is needed because WSClient has a default 100 ms notification delay before updates
-                    setTimeout(() => {
-                        showBoard(props.boards[nextBoardId as number].id)
-                    }, 120)
+                if (!targetBoardId) {
+                    for (const category of props.allCategories) {
+                        for (const m of category.boardMetadata) {
+                            if (m.boardID === deleteBoard.id) {
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply webapp/src/components/sidebar/sidebarCategory.tsx:210]: The TypeScript type is `boardMetadata: CategoryBoardMetadata[]`, non-optional, no null in the type definition. The store always initialises it as [].

[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply webapp/src/components/sidebar/sidebarCategory.tsx:210]: We can resolve this one.

**Review by harshilsharma63 — APPROVED** [review]
